app/src/main/java/com/aptoide/uploader/account/AutoLoginManager.java

```diff
@@ -6,56 +6,113 @@
+  }
+
+  public Single<AutoLoginCredentials> getFirstStoredUserCredentials() {
     try {
       String URL = "content://cm.aptoide.pt.StubProvider";
       Uri token_uri = Uri.parse(URL + "/token");
       Uri refresh_token_uri = Uri.parse(URL + "/refreshToken");
       Uri repo_uri = Uri.parse(URL + "/repo");
+      Uri email_uri = Uri.parse(URL + "/loginName");
+      Uri nickname_uri = Uri.parse(URL + "/loginNickname");
+      Uri avatar_uri = Uri.parse(URL + "/loginAvatar");
 
       Cursor c1 = context.getContentResolver()
           .query(token_uri, null, null, null, null);
       Cursor c2 = context.getContentResolver()
           .query(refresh_token_uri, null, null, null, null);
       Cursor c3 = context.getContentResolver()
           .query(repo_uri, null, null, null, null);
+      Cursor c4 = context.getContentResolver()
+          .query(email_uri, null, null, null, null);
+      Cursor c5 = context.getContentResolver()
+          .query(nickname_uri, null, null, null, null);
+      Cursor c6 = context.getContentResolver()
+          .query(avatar_uri, null, null, null, null);
 
-      if (c1 != null && c2 != null && c3 != null) {
-
+      if (c1 != null && c2 != null) {
         c1.moveToFirst();
         c2.moveToFirst();
-        c3.moveToFirst();
-
-        AutoLoginCredentials autoLoginCredentials = new AutoLoginCredentials();
         autoLoginCredentials.setAccessToken(c1.getString(c1.getColumnIndex("userToken")));
         autoLoginCredentials.setRefreshToken(c2.getString(c2.getColumnIndex("userRefreshToken")));
-
         c1.close();
         c2.close();
-        c3.close();
 
-        return Single.just(autoLoginCredentials);
+        if (c3 != null) {
+          c3.moveToFirst();
+          autoLoginCredentials.setStoreName(c3.getString(c3.getColumnIndex("userRepo")));
+          c3.close();
+        }
+
+        if (c4 != null) {
+          c4.moveToFirst();
+          autoLoginCredentials.setEmail(c4.getString(c4.getColumnIndex("loginName")));
+          c4.close();
+        }
+        if (c5 != null) {
+          c5.moveToFirst();
+          autoLoginCredentials.setName(c5.getString(c5.getColumnIndex("loginNickname")));
+          c5.close();
+        }
+        if (c6 != null) {
+          c6.moveToFirst();
+          autoLoginCredentials.setAvatarPath(c6.getString(c6.getColumnIndex("loginAvatar")));
+          c6.close();
+        }
       }
     } catch (Exception e) {
       e.printStackTrace();
       return Single.error(e);
     }
-    return Single.just(new AutoLoginCredentials());
+    return Single.just(autoLoginCredentials);
   }
 
-  public boolean getAutologinFlag() {
-    return persistence.isForcedLogout();
+  public boolean isNullOrEmpty(String str) {
+    return str == null || str.trim()
+        .isEmpty();
   }
 
-  public void setAutoLoginFlag(boolean flag) {
-    persistence.setForcedLogout(flag);
+  public void checkAvatar(String name, Navigator nav) {
```

Comment:
I first implemented it in each presenter but resulted in duplicated code (in 3 presenters) and since that are 7 possible combinations for this navigation, i'm not sure how to this in each presenter without duplication of code